src/gpt-auto-generator/gpt-auto-generator.c

```diff
@@ -514,6 +516,51 @@ static bool slash_boot_exists(void) {
         return (cache = false);
 }
 
+static bool fstab_is_same_device(const char *where, const char *what) {
```

Comment:
I think this is an extension of `fstab_is_mount_point()`. How about to rename `fstab_is_mount_point_full()` and move to fstab-util.c ?